### PR TITLE
PLAN-502 Password policy tooltips

### PR DESCRIPTION
**Pull request by lastminutediorama**
We decided to pause on what exactly our password requirements would say, so this is just a draft PR with an example of a popup that appears on password field focus. Since we're probably not looking to merge this yet, I figured I'd request feedback on the UX and implementation.

pwhint

### CODE REVIEW

File: src/interface/src/app/password-reset/password-reset.component.html [hunk 1]
```diff
@@ -49,6 +59,19 @@ <h1 class="title">Reset Password</h1>
     </form>
   </div>
 
+  <mat-menu class="password-reqs-menu" #popoverMenu="matMenu">
```
Comment (lastminutediorama) [inline src/interface/src/app/password-reset/password-reset.component.html:62]: Is there a preferred way to DRY this up?

**Comment by pflopez** [comment]
I like this! I think looks great and its super clear, and at least on chrome it plays nicely with the saved passwords module
Screen Shot 2023-10-30 at 12 14 32


I did find a couple of issues with the password confirmation field.
- This field shouldn't be marked as error until you actually type something...or field the first password:

Screen Shot 2023-10-30 at 12 15 15

Also once I fill the password, but mismatch the password confirmation, that field is not marked as error

Screen Shot 2023-10-30 at 12 15 30

File: src/interface/src/app/password-reset/password-reset.component.html (hunk 1 above)
Comment (pflopez) [reply src/interface/src/app/password-reset/password-reset.component.html:62]: hm good question, thing is you need the reference for matMenuTriggerFor, so not sure how you can move it to a component.

Comment (pflopez) [reply src/interface/src/app/password-reset/password-reset.component.html:62]: I mean you can move the contents to a component and keep something like
```
<mat-menu class="password-reqs-menu" #popoverMenu="matMenu">
  <app-pass-reqs/>
</mat-menu>
```

Comment (lastminutediorama) [reply src/interface/src/app/password-reset/password-reset.component.html:62]: Right, I considered it, but it felt like overkill to create a new component. On the other hand, I don't love having this sort of critical, user-facing text across multiple files...I'm not sure how to handle this either, unless we wanted to implement a central strings collection, like we might have for i18n.

**Comment by lastminutediorama** [comment]
> I did find a couple of issues with the password confirmation field.
> * This field shouldn't be marked as error until you actually type something...or field the first password:

Yes, I agree. I wrote a validation function for the Change Password dialog that handles this a little differently, and I want to make that available to the signup and reset password components. I can probably do that today.

**Pull request closed without merging**